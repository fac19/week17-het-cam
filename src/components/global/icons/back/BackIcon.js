import React from "react";
import { useHistory } from "react-router-dom";

import { makeStyles } from "@material-ui/core/styles";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";

import styled from "styled-components";
import { linkColour } from "../../colours";

const useStyles = makeStyles((theme) => ({
  root: {
    height: "100%",
    color: "white",
  },
}));

const BackButton = styled.button`
  border-radius: 100%;
  border: none;
  background-color: ${linkColour};
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
`;

const BackIcon = (props) => {
  let history = useHistory();
  const classes = useStyles();
  return (
    <BackButton
      aria-label="Go back to previous page"
      onClick={() => history.push(props.to)}>
      <ArrowBackIcon className={classes.root} />
    </BackButton>
  );
};

export default BackIcon;
