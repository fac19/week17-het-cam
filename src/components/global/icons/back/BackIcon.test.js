import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import BackIcon from "./BackIcon";

test("BackIcon component is styled", () => {
  const tree = renderer.create(<BackIcon />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree).toHaveStyleRule("height", "2.5rem");
});
