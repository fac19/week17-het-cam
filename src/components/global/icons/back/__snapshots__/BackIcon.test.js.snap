// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackIcon component is styled 1`] = `
.c0 {
  border-radius: 100%;
  border: none;
  background-color: #026CB6;
  width: 2.5rem;
  height: 2.5rem;
  cursor: pointer;
}

<button
  aria-label="Go back to previous page"
  className="c0"
  onClick={[Function]}
>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root makeStyles-root-1"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
    />
  </svg>
</button>
`;
